SU2_CFD/include/numerics/turbulent/turb_sources_new.hpp

```diff
@@ -0,0 +1,662 @@
+        Jacobian_i[0] += cb1 * (ScalarVar_i[0] * dShat + Shat) * Volume;
+      }
+
+      /*--- Implicit part, destruction term ---*/
+
+      dr = (Shat - ScalarVar_i[0] * dShat) * inv_Shat * inv_Shat * inv_k2_d2;
+      if (r == 10.0) dr = 0.0;
+      dg = dr * (1. + cw2 * (6.0 * pow(r, 5.0) - 1.0));
+      dfw = dg * glim * (1. - g_6 / (g_6 + cw3_6));
+      Jacobian_i[0] -= cw1 * (dfw * ScalarVar_i[0] + 2.0 * fw) * ScalarVar_i[0] / dist_i_2 * Volume;
+
+      ft2_class::get(&ft2, &d_ft2);
+
+      su2double pingu = ft2 * 3.14;
+
+      ModVort_class::get();
+    }
+  }
+};
+
+/*------------------------------------------------------------------------------
+| Compute the ft2-term and its derivative
+| * \param[in] ct3 - constant ct3
+| * \param[in] ct4 - constant ct4
+| * \param[in] nue - nu_tilde
+| * \param[in] nul - dynamic laminar viscosity
+| * \param[out] ft2 - ft2 variable
+| * \param[out] d_ft2 - derivative of ft2 w.r.t. nue variable
+------------------------------------------------------------------------------*/
+
+/*!
+ * \brief SU2 baseline ft2 term value and its derivative. ft2=0.0
+ */
+template <class Base>
+class ft2_Bsl : public Base {
+  static double get(su2double* ft2, su2double* d_ft2) {
+    *ft2 = 0.0;
+    *d_ft2 = 0.0;
+  }
+};
+
+/*!
+ * \brief non-zero ft2 term according to the literature and its derivative.
+ */
+template <class Base>
+class ft2_nonzero : public Base {
+  using Base::ct3;
+  using Base::ct4;
+  using Base::d_xsi;
+  using Base::xsi;
+
+  static void get(su2double* ft2, su2double* d_ft2) {
+    const su2double xsi2 = xsi * xsi;
+
+    *ft2 = ct3 * exp(-ct4 * xsi2);
+    *d_ft2 = -2.0 * ct4 * xsi * *ft2 * d_xsi;
+  }
+};
+
+/* The above causes a cyclic dependency between the turbulence base and the ft2 and co. classes.
+ * To break that you need to template the get method instead of the class, for example: */
+struct ft2_nonzero {
+  template <class Base>
+  static void get(const Base& base, su2double& ft2, su2double& d_ft2) {
+    const su2double xsi2 = pow(base.xsi, 2);
+    ft2 = base.ct3 * exp(-base.ct4 * xsi2);
+    d_ft2 = -2.0 * base.ct4 * base.xsi * ft2 * base.d_xsi;
+  }
+};
+/* Now you pass the numerics class itself to "get" to access the member variables,
+ * which you'll have to make public, for example in ComputeResidual:
+ * ft2_class::get(*this, ft2, d_ft2);
+ *
+ * This is not a perfect solution because for performance we should cut down on "aux" class variables.
+ * An alternative is to put such variables in a struct, which can then be a local variable in ComputeResidual. */
+struct CommonVariables {
+  su2double ft2, d_ft2, r, d_r, g, d_g, g_6, glim, fw, Ji, Ji_2, Ji_3, d_Ji, S, Omega, Shat, d_Shat, inv_Shat, fv1, fv2;
+};
+/*
+ *  ResidualType<> ComputeResidual(const CConfig* config) final {
+ *    CommonVariables data;
+ *    ...
+ *    data.Ji = ...
+ *    ...
+ *    ft2_class::get(data, ft2, d_ft2);
+ */
+/* This way you can also dispense with templating "get". */
```

Comment:
I think this will work.

Reply:
As @pcarruscag indicated: The structure I have implemented in  #3dd64f44e37b8b4ada9a31c5d450b722d3ef8554 for the turb_sources.hpp causes a cyclic dependency between the turbulence base and the variants classes. To break that one needs to template the `get` method in the variants classes instead of the class, for example the procedure he suggested in #2f60dbb4cae19aa15b9b6578043838f8eefd39b0